lib/Frontend/ParseableInterfaceSupport.cpp

```diff
@@ -188,6 +189,8 @@ swiftModuleIsUpToDate(clang::vfs::FileSystem &FS,
     return false;
 
   for (auto In : AllDeps) {
+    if (OuterTracker)
+      OuterTracker->addDependency(In.Path, /*IsSystem=*/false);
```

Comment:
I guess we ought to be tracking system-ness too.

Reply:
Yeah I meant to ask about that. The existing DependencyTracker can't provide this bit since it just forwards through to Clang's dependency tracker, which also provides no interface for it, and in fact doesn't even store it -- it's only an in-flight flag that controls whether or not a dependency gets _added_, but it's not recorded anywhere whether each dependency _originated_ as a system or non-system entry. So I'd need to extend the interfaces and implementations of .. both? Seems a bit heavy. I'm not sure what else to do here.

Reply:
Hm. Let's file a bug to come back to it and think about it later.

Reply:
Ok: rdar://45957588